src/main/java/org/broadinstitute/dsm/model/ParticipantWrapper.java

```diff
@@ -246,6 +246,36 @@ else if (DBConstants.DDP_ABSTRACTION_ALIAS.equals(source)) {
         }
     }
 
+    public static Map<String, Map<String, Object>> getParticipantESDataConsideringNumberOfParameters(@NonNull DDPInstance instance, Map<String, String> filters, String source) {
+        Map<String, Map<String, Object>> participantESData;
+        String wholeFilter = filters.get(source);
+        String[] betweenAnds = wholeFilter.split(Filter.AND_TRIMMED);
+        boolean tooManyParameters = false;
+        participantESData = new HashMap<>();
+        String lessThanLimit = Arrays.stream(betweenAnds).filter(query -> query.split(Filter.OR_TRIMMED).length <= 900).collect(Collectors.joining(Filter.AND));
+        for (String query: betweenAnds) {
+            String[] orQueries = query.split(Filter.OR_TRIMMED);
+            if (orQueries.length <= 900) {
+                continue;
+            }
+            tooManyParameters = true;
```

Comment:
looks like it will be always true

Reply:
It will be false if none of the between AND-s groups have too many parameters

Reply:
:doh: I missed the continue in the if
Sorry!